papers/oleksandr_pavlyk/numba/svml.rst

```diff
@@ -29,23 +29,102 @@ For example, the argument to the ``_mm_sin_ps`` intrinsic is a packed 128-bit ve
     
-    # IR generated
+    label 16:
+        $16.2 = iternext(value=$phi16.1)         ['$16.2', '$phi16.1']
+        $16.3 = pair_first(value=$16.2)          ['$16.2', '$16.3']
+        $16.4 = pair_second(value=$16.2)         ['$16.2', '$16.4']
+        del $16.2                                []
+        $phi19.1 = $16.3                         ['$16.3', '$phi19.1']
+        del $16.3                                []
+        branch $16.4, 19, 48                     ['$16.4']
+    label 19:
+        del $16.4                                []
+        i = $phi19.1                             ['$phi19.1', 'i']
+        del $phi19.1                             []
+        $19.2 = global(math: <module 'math' from '/path_stripped/lib-dynload/math.cpython-35m-x86_64-linux-gnu.so'>) ['$
+    19.2']
+        $19.3 = getattr(attr=sin, value=$19.2)   ['$19.2', '$19.3']
+        del $19.2                                []
+        $19.6 = getitem(index=i, value=x)        ['$19.6', 'i', 'x']
+        $19.7 = call $19.3($19.6)                ['$19.3', '$19.6', '$19.7']
+        del $19.6                                []
+        del $19.3                                []
+        y[i] = $19.7                             ['$19.7', 'i', 'y']
+        del i                                    []
+        del $19.7                                []
+        jump 16                                  []
 
-.. code-block:: Asm
     
-    # generated code
+We can see direct call into svml library:
 
+.. code-block:: Asm
+    
+            movq    %rdi, 8(%rsp)
+            movq    %r13, 16(%rsp)
+            movq    %r15, 24(%rsp)
+            subq    %rbx, %r12
+            leaq    96(%rdx), %r14
+            leaq    96(%rsi), %r15
+            movabsq $__svml_sin4_ha, %rbp
+            movq    %rbx, %r13
+            .p2align        4, 0x90
+    .LBB0_13:
+            vmovups -96(%r14), %ymm0
+            vmovups -64(%r14), %ymm1
+            vmovups %ymm1, 32(%rsp)
+            vmovups -32(%r14), %ymm1
+            vmovups %ymm1, 64(%rsp)
+            vmovups (%r14), %ymm1
+            vmovups %ymm1, 128(%rsp)
+            callq   *%rbp
+            vmovups %ymm0, 96(%rsp)
+            vmovups 32(%rsp), %ymm0
+            callq   *%rbp
+            vmovups %ymm0, 32(%rsp)
+            vmovups 64(%rsp), %ymm0
+            callq   *%rbp
+            vmovups %ymm0, 64(%rsp)
+            vmovupd 128(%rsp), %ymm0
+            callq   *%rbp
+            vmovups 96(%rsp), %ymm1
+            vmovups %ymm1, -96(%r15)
+            vmovups 32(%rsp), %ymm1
+            vmovups %ymm1, -64(%r15)
+            vmovups 64(%rsp), %ymm1
+            vmovups %ymm1, -32(%r15)
+            vmovupd %ymm0, (%r15)
+            subq    $-128, %r14
+            subq    $-128, %r15
+            addq    $-16, %r13
+            jne     .LBB0_13
+
+So having enabled svml high accuracy functions this jitted code can work more than 4x times faster than with usual libm's code.
```

Comment:
Replace this sentence:

'Thanks to enabled support of high accuracy SVML functions in LLVM this jitted code sees more than 4x increase in performance'